lib/chef/knife/ssh.rb

```diff
@@ -219,23 +219,28 @@ def search_nodes
       # @return [Hash<Symbol, Object>]
       def session_options(host, port, user=nil)
         ssh_config = Net::SSH.configuration_for(host)
-        {}.tap do |opts|
+        ssh_config.tap do |opts|
```

Comment:
For example, in exist spec.

https://github.com/chef/chef/blob/master/spec/unit/knife/ssh_spec.rb#L189-L194

`:timeout` is set, but it'll be ignore. isn't it?